catalog/dags/data_refresh/create_filtered_index_dag.py

```diff
@@ -0,0 +1,230 @@
+            check_existence=False,
+            execution_date_fn=lambda _: get_most_recent_dag_run(data_refresh_dag_id),
+            mode="reschedule",
+        )
+        try:
+            wait_for_filtered_index_creation.execute(context)
+        except AirflowSensorTimeout:
+            raise ValueError(
+                f"{media_type} data refresh concurrency check failed. "
+                "Filtered index creation cannot start during a data refresh."
+            )
+
+    @task()
+    def generate_index_suffix(default_suffix: str):
+        return default_suffix or uuid.uuid4().hex
+
+    def create_and_populate_filtered_index(
+        origin_index_suffix: str | None,
+        destination_index_suffix: str | None,
+    ):
+        create_payload = {}
+        if origin_index_suffix:
+            create_payload["origin_index_suffix"] = origin_index_suffix
+        if destination_index_suffix:
+            create_payload["destination_index_suffix"] = destination_index_suffix
+
+        return ingestion_server.trigger_and_wait_for_task(
+            action="CREATE_AND_POPULATE_FILTERED_INDEX",
+            model=media_type,
+            data=create_payload or None,
+            timeout=data_refresh.create_filtered_index_timeout,
+        )
+
+    def point_alias(destination_index_suffix: str):
+        point_alias_payload = {
+            "alias": target_alias,
+            "index_suffix": f"{destination_index_suffix}-filtered",
+        }
+
+        return ingestion_server.trigger_task(
+            action="POINT_ALIAS",
+            model=media_type,
+            data=point_alias_payload,
+        )
+
+    with DAG(
+        dag_id=f"create_filtered_{media_type}_index",
+        default_args=DAG_DEFAULT_ARGS,
+        schedule=None,
+        start_date=datetime(2023, 4, 1),
+        tags=["data_refresh"],
+        max_active_runs=1,
+        catchup=False,
+        doc_md=__doc__,
+        params={
+            "force": Param(
+                default=False,
+                type="boolean",
+                description=(
+                    "Bypass data refresh concurrency check. "
+                    "Should only ever be used by the data_refresh "
+                    "DAGs when triggering filtered index creation "
+                    "at the end of a data refresh. This should not "
+                    "be used when manually running this DAG "
+                    "unless you're absolutely sure of what you're "
+                    "doing. This check exists to prevent race "
+                    "conditions and should not be ignored lightly."
+                ),
+            ),
+            "origin_index_suffix": Param(
+                default=None,
+                type=["string", "null"],
+                description=(
+                    "See https://github.com/WordPress/openverse/blob/7427bbd4a8178d05a27e6fef07d70905ec7ef16b/ingestion_server/ingestion_server/indexer.py#L495-L497. "  # noqa: E501
```

Comment:
Weirdly, referencing a permalink seems odd considering that it might not reflect the present situation of the ingestion server, but at the same time, referencing `main` would be worse as it might start pointing to something else. Is there a way to not reference the ingestion server code and explain in a comment?

Reply:
There are two other options, in my opinion:

1. Repeat the documentation here
  - Pros: Does not reference a static part of the ingestion server code
  - Cons: It will get out of date probably almost as soon as it is committed, as these things go
2. Rather than link to the code, just say "see `Indexer::create_and_populate_filtered_index` documentation in ingestion server"
  - Pros: Never goes out of date unless the method name changes (which would force changes in this DAG as well, because the method name is concretely tied to the task name that the DAG must know
  - Cons: Less convenient

I'd rather do the second if these permalinks aren't to satisfaction.

Reply:
Though, I will say, ideally we'd have task documentation either statically written in the ingestion server documentation or pulled from docstrings in the code.